### PR TITLE
Fixed #19215 -- Fixed rendering ClearableFileInput when editing with invalid files.

### PR DESCRIPTION
**Pull request by mgaligniana**
Hello!

Thanks Mariusz for Cc'me in this [ticket](https://code.djangoproject.com/ticket/19215)!

I just applied the [mcardillo55's fix](https://github.com/django/django/pull/600/commits/69808dfaf9c96b0a60121671ab7d0dac7d625b56) + added a selenium test. In [his pull request](https://github.com/django/django/pull/600) I didn't see any review/comment/change required to apply.

To run the test please do: `./runtests.py admin_widgets.tests.ImageFieldWidgetsSeleniumTests --selenium=<chrome|firefox|safari> --parallel=1`

Thanks!

### CODE REVIEW

File: tests/files/test.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+test
```
Comment (mgaligniana) [inline tests/files/test.txt:1]: Add this new file to use it in the test case is OK? Or should I create a temporary file?

File: tests/admin_widgets/models.py [hunk 2]
```diff
@@ -199,3 +199,8 @@ class Profile(models.Model):
 
     def __str__(self):
         return self.user.username
+
+
+class Book(models.Model):
+    title = models.CharField(max_length=255)
+    cover = models.ImageField(upload_to="images", blank=True, null=True)
```
Comment (mgaligniana) [inline tests/admin_widgets/models.py:206]: This creates the new `images/` directory in my local. How should I manage it to make it temporary?

File: tests/admin_widgets/tests.py [hunk 3]
```diff
@@ -1767,3 +1768,58 @@ def test_ForeignKey_using_to_field(self):
...
+        title_input = self.selenium.find_element(By.ID, "id_title")
+        title_input.send_keys("Book title")
+        image_input = self.selenium.find_element(By.ID, image_input_id)
+        image_input.send_keys(f"{tests_files_folder}/test.png")
+        _submit_and_wait()
+        book = Book.objects.last()
+        self.assertEqual(book.title, "Book title")
+        self.assertEqual(book.cover.name, "images/test.png")
+
+        # Upload file is not supported by Safari with Selenium, so instead of upload a
```
Comment (mgaligniana) [inline tests/admin_widgets/tests.py:1803]: Seems Safari doesn't support file upload via Selenium, please see: https://stackoverflow.com/a/62920889

File: tests/admin_widgets/tests.py [hunk 4]
```diff
@@ -1767,3 +1768,58 @@ def test_ForeignKey_using_to_field(self):
...
+        # non image file upload a broken one to get the same error behaviour.
```
Comment (mgaligniana) [inline tests/admin_widgets/tests.py:1804]: Seems Safari doesn't support file upload via Selenium, please see: https://stackoverflow.com/a/62920889

**Comment by mgaligniana** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
> The `pull-requests-selenium/database=postgres,label=focal-pr,python=python3.10` buildot fails because of
> 
> image

This test became extremely flaky few weeks ago :disappointed: Would you like to try to improve it?

**Comment by mgaligniana** [comment]
buildbot, test on selenium.

**Comment by mgaligniana** [comment]
Done! @felixxm

**Comment by felixxm** [comment]
> Done! @felixxm

Many thanks! Merged in 1d6948096f6fe7aa887d651e01e9af8e4ef349a2.

**Review by felixxm — COMMENTED** [review]
@mgaligniana Thanks :+1: I pushed small edits.

**Comment by felixxm** [comment]
@mcardillo55 Thanks for the initial patch :+1:

**Comment by felixxm** [comment]
buildbot, test on selenium.

File: tests/admin_widgets/tests.py [hunk 5]
```diff
@@ -1772,3 +1772,56 @@ def test_ForeignKey_using_to_field(self):
...
+        title_input = self.selenium.find_element(By.ID, "id_name")
+        title_input.send_keys("Joe Doe")
+        photo_input = self.selenium.find_element(By.ID, photo_input_id)
+        photo_input.send_keys(f"{tests_files_folder}/test.png")
+        _submit_and_wait()
+        student = Student.objects.last()
+        self.assertEqual(student.name, "Joe Doe")
+        self.assertEqual(student.photo.name, "photos/test.png")
+        # Uploading non-image files is not supported by Safari with Selenium,
+        # so uploade a broken one instead.
```
Comment (mgaligniana) [inline tests/admin_widgets/tests.py:1807]:
```suggestion
        # so upload a broken one instead.
```

Comment (felixxm) [reply tests/admin_widgets/tests.py:1807]: Good catch :dart:

**Comment by felixxm** [comment]
Merged in c0fc1b5302f5d9d82e2255fb5758321fbac34949.

**Pull request closed without merging**